frontend/searching.cpp

```diff
@@ -0,0 +1,79 @@
+#include "mainwindow.h"
+#include <tuple>
+
+// Search table data
+void MainWindow::on_searchBox_textChanged(const QString &arg1)
+{
+    foundIndices.clear();
+    currentFoundIndex = 0;
+    if (arg1 == "")
+    {
+        ui->runDataTable->selectionModel()->clearSelection();
+        return;
+    }
+    // Find all occurences of search string in table elements
+    for (int i = 0; i < proxyModel->rowCount(); i++)
```

Comment:
```suggestion
    for (auto i = 0; i < proxyModel->rowCount(); ++i)
```
Prefer prefix (`++i`) over postfix (`i++`) increment, since the former is fewer operations. Utterly meaningless performance-wise in a loop like this, but a good habit to get in to!